### PR TITLE
chore(workflow): Bump setup-node version in our workflows

### PR DESCRIPTION
**Pull request by ioanabrooks**
#### Description of changes

While running build we are experiencing node setup failures such as:

```
Download action repository 'actions/setup-node@v2' (SHA:1f8c6b94b26d0feae1e387ca63ccbdc44d27b561)
Warning: Failed to download action 'https://api.github.com/repos/actions/setup-node/tarball/1f8c6b94b26d0feae1e387ca63ccbdc44d27b561'. Error: Response status code does not indicate success: 500 (Internal Server Error).
Warning: Back off 23.799 seconds before retry.
Warning: Failed to download action 'https://api.github.com/repos/actions/setup-node/tarball/1f8c6b94b26d0feae1e387ca63ccbdc44d27b561'. Error: Response status code does not indicate success: 500 (Internal Server Error).
Warning: Back off 11.876 seconds before retry.
Error: Response status code does not indicate success: 500 (Internal Server Error).
```

Bumping to latest github node-setup action which has caching and better retry logic.

#### Description of how you validated changes

PoC worflow: https://github.com/aws-amplify/amplify-ui/actions/runs/3146684910/jobs/5115410248

```

Run actions/setup-node@v3
[8](https://github.com/aws-amplify/amplify-ui/actions/runs/3146684910/jobs/5115410248#step:3:9)
Attempt to resolve LTS alias from manifest...
[9](https://github.com/aws-amplify/amplify-ui/actions/runs/3146684910/jobs/5115410248#step:3:10)
Found in cache @ /opt/hostedtoolcache/node/16.17.0/x64
[10](https://github.com/aws-amplify/amplify-ui/actions/runs/3146684910/jobs/5115410248#step:3:11)

[11](https://github.com/aws-amplify/amplify-ui/actions/runs/3146684910/jobs/5115410248#step:3:12)
/usr/local/bin/yarn --version
[12](https://github.com/aws-amplify/amplify-ui/actions/runs/3146684910/jobs/5115410248#step:3:13)
1.22.19
[13](https://github.com/aws-amplify/amplify-ui/actions/runs/3146684910/jobs/5115410248#step:3:14)
/usr/local/bin/yarn cache dir
[14](https://github.com/aws-amplify/amplify-ui/actions/runs/3146684910/jobs/5115410248#step:3:15)
/home/runner/.cache/yarn/v6
[15](https://github.com/aws-amplify/amplify-ui/actions/runs/3146684910/jobs/5115410248#step:3:16)
Received 159383552 of 799675467 (19.9%), 151.2 MBs/sec
[16](https://github.com/aws-amplify/amplify-ui/actions/runs/3146684910/jobs/5115410248#step:3:17)
Received 373293056 of 799675467 (46.7%), 177.5 MBs/sec
[17](https://github.com/aws-amplify/amplify-ui/actions/runs/3146684910/jobs/5115410248#step:3:18)
Received 553648128 of 799675467 (69.2%), 175.5 MBs/sec
[18](https://github.com/aws-amplify/amplify-ui/actions/runs/3146684910/jobs/5115410248#step:3:19)
Received 759169024 of 799675467 (94.9%), 180.5 MBs/sec
[19](https://github.com/aws-amplify/amplify-ui/actions/runs/3146684910/jobs/5115410248#step:3:20)
Received 799675467 of 799675467 (100.0%), 153.8 MBs/sec
[20](https://github.com/aws-amplify/amplify-ui/actions/runs/3146684910/jobs/5115410248#step:3:21)
Cache Size: ~763 MB (799675467 B)
[21](https://github.com/aws-amplify/amplify-ui/actions/runs/3146684910/jobs/5115410248#step:3:22)
/usr/bin/tar --use-compress-program unzstd -xf /home/runner/work/_temp/32b3e694-ff5d-4ea9-a5b7-bcdc40bd87b0/cache.tzst -P -C /home/runner/work/amplify-ui/amplify-ui
[22](https://github.com/aws-amplify/amplify-ui/actions/runs/3146684910/jobs/5115410248#step:3:23)
Cache restored successfully
[23](https://github.com/aws-amplify/amplify-ui/actions/runs/3146684910/jobs/5115410248#step:3:24)

[24](https://github.com/aws-amplify/amplify-ui/actions/runs/3146684910/jobs/5115410248#step:3:25)
Cache restored from key: node-cache-Linux-yarn-1e0437425288b07b35fe39ffc1a47926aee4a6d433d20316c8a556b2ae7ecfa2

```

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/poc-build-and-runtime-test.yml [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# Description: this workflow runs build and simple runtime tests on supported
+#              frameworks and their build systems.
+#
+# Triggered by: it runs every 15 minutes.
+
+name: PoC Build and Runtime Test
+
+on:
+  pull_request:
+    - branches: [main]
```
Comment (wlee221) [inline .github/workflows/poc-build-and-runtime-test.yml:10]:
```suggestion
  pull_request:
    branches: [main]
```

Comment (wlee221) [reply .github/workflows/poc-build-and-runtime-test.yml:10]: fyi https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet

Comment (ioanabrooks) [reply .github/workflows/poc-build-and-runtime-test.yml]: According to github [docs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions) actually:
```
on:
  pull_request:
    branches:
      - main
  ```

Comment (wlee221) [reply .github/workflows/poc-build-and-runtime-test.yml]: Yeah, there're multiple syntax to do it. Either works  😈

**Review by wlee221 — APPROVED** [review]
🚢

**Review by hbuchel — APPROVED** [review]

**Pull request merged**